src/main/java/com/ably/kafka/connect/mapping/DefaultMessageSinkMapping.java

```diff
@@ -35,5 +40,23 @@ public Message getMessage(SinkRecord record) {
         return message;
     }
 
+    private Message messageFromRecord(SinkRecord record) {
+        final String messageName = configValueEvaluator.evaluate(record, sinkConnectorConfig.getString(MESSAGE_CONFIG));
+
+        if (record.valueSchema() == null) {
+            return new Message(messageName, record.value());
+        }
+
+        final Schema valueSchema = record.valueSchema();
+        switch (valueSchema.type()) {
```

Comment:
I kept this switch especially to also consider special processing of other schema types. Copied from source, these are all possible types for it 
`INT8,  INT16,INT32, INT64, FLOAT32, FLOAT64,BOOLEAN,STRING, BYTES,ARRAY,MAP,STRUCT`
Any other type than struct - the ably message will directly be built . I did not include these in this PR as other cases than STRUCT were not the case for Avro

Reply:
I have converted switch to an if statement 3d5084998a29708b86ba24f6f01f2a153668a4aa

Reply:
Thanks. https://github.com/ably/kafka-connect-ably/commit/3d5084998a29708b86ba24f6f01f2a153668a4aa specifically addresses my question around "`switch` for a single case".

However, what it doesn't address is my second concern, where I question the validity of your code path if the type is anything but `STRUCT` here... e.g. for `BYTES`, or even `MAP`. I would still expect to see an exception to be thrown here in those cases, or am I missing something?

Reply:
Sorry @QuintinWillison I could have been a bit more clear on explanation

> I did not include these in this PR as other cases than STRUCT were not the case for Avro
Short answer is outside of scope of this PR. 

In case of all other types the message will be passed exactly as they are which was the case before this change. But I think we should address this in subsequent PRs potentially considering Jsonfiying them (TBD)